src/Files.App/Actions/Content/Selection/SelectAllAction.cs

```diff
@@ -23,22 +23,25 @@ public bool IsExecutable
 		{
 			get
 			{
-				if (context.PageType is ContentPageTypes.Home)
-					return false;
-
 				var page = context.ShellPage;
+				 
+				int itemCount = page.FilesystemViewModel.FilesAndFolders.Count;
+				int selectedItemCount = context.SelectedItems.Count;
+
 				if (page is null)
 					return false;
 
-				int itemCount = page.FilesystemViewModel.FilesAndFolders.Count;
-				int selectedItemCount = context.SelectedItems.Count;
+				if (context.PageType is ContentPageTypes.Home)
+					return false;
+				
```

Comment:
As is, a NullReferenceException will be thrown if `page` is null. These changes are no longer necessary.

Reply:
Do you mean that I should revert these changes by placing the declaration of isCommandPaletteOpen before var page = context.ShellPage; as it was previously? I need more clarity on what needs moved. Thanks.

Reply:
I mean these changes (L26-L36) should be revert to the same code as the current main branch.

Reply:
Thank you. This means that none of the actions will have bool isCommandPaletteOpen code. I will revert all three actions to main branch state. Is this correct?

Reply:
No, no, I mean that only this part should be reverted to the main branch status. `isCommandPaletteOpen` should remain.